### PR TITLE
Docs: Add docs for caching timeouts

### PR DESCRIPTION
**Pull request by toddtreece**
**What this PR does / why we need it**:

New cache timeout settings were added recently, so this adds the related documentation.

**Which issue(s) this PR fixes**:

Adds docs for:

* https://github.com/grafana/grafana-enterprise/issues/1915
* https://github.com/grafana/grafana-enterprise/issues/1927

**Special notes for your reviewer**:

### CODE REVIEW

File: docs/sources/enterprise/enterprise-configuration.md [hunk 1]
```diff
@@ -353,14 +353,40 @@ This value is `true` by default.
 
 ### ttl
 
-_Time to live_ (TTL) is the time that a query result is stored in the caching system before it is deleted or refreshed. This setting defines the time to live for query caching, when TTL is not configured in data source settings. The default value is `5m` (5 minutes).
+_Time to live_ (TTL) is the time that a query result is stored in the caching system before it is deleted or refreshed. This setting defines the time to live for query caching, when TTL is not configured in data source settings. The default value is `1m` (1 minute).
```
Comment (toddtreece) [inline docs/sources/enterprise/enterprise-configuration.md:356]: @kminehart just to confirm this change, is this default set to `1m` here? https://github.com/grafana/grafana-enterprise/blob/ea5b5428f9dd8ca336dc53ce9eeb57b17d1eb587/src/pkg/extensions/caching/config/storage_settings.go#L26

File: docs/sources/enterprise/enterprise-configuration.md [hunk 2]
```diff
@@ -353,14 +353,40 @@ This value is `true` by default.
...
+
+### max_ttl_ms
```
Comment (toddtreece) [inline docs/sources/enterprise/enterprise-configuration.md:358]: @osg-grafana i'm going to update the setting parsing logic to match how `ttl` handles durations. i'll leave this PR in draft and update the docs once the related change merges

File: docs/sources/enterprise/enterprise-configuration.md (hunk 1 above)
Comment (kminehart) [reply docs/sources/enterprise/enterprise-configuration.md:356]: yes that's where the default is set. I think it was originally set to 5m but I figured that was too much.

**Review by stephaniehingtgen — APPROVED** [review]

File: docs/sources/enterprise/enterprise-configuration.md [hunk 3]
```diff
@@ -353,14 +353,44 @@ This value is `true` by default.
 
 ### ttl
 
-_Time to live_ (TTL) is the time that a query result is stored in the caching system before it is deleted or refreshed. This setting defines the time to live for query caching, when TTL is not configured in data source settings. The default value is `5m` (5 minutes).
+_Time to live_ (TTL) is the time that a query result is stored in the caching system before it is deleted or refreshed. This setting defines the time to live for query caching, when TTL is not configured in data source settings. The default value is `1m` (1 minute).
+
+### max_ttl
+
+The max duration that a query result is stored in the caching system before it is deleted or refreshed. This value will override `ttl` config option or data source setting if the `ttl` value is greater than `max_ttl`. To disable this timeout, set this value to `0s`.
```
Comment (stephaniehingtgen) [inline docs/sources/enterprise/enterprise-configuration.md:360]:
```suggestion
The max duration that a query result is stored in the caching system before it is deleted or refreshed. This value will override `ttl` config option or data source setting if the `ttl` value is greater than `max_ttl`. To disable this constraint, set this value to `0s`.
```

File: docs/sources/enterprise/enterprise-configuration.md [hunk 4]
```diff
@@ -353,14 +353,44 @@ This value is `true` by default.
...
+
+The default is `0s` (disabled).
+
+> **Note:** Disabling this timeout is not recommended in production environments.
```
Comment (stephaniehingtgen) [inline docs/sources/enterprise/enterprise-configuration.md:364]:
```suggestion
> **Note:** Disabling this constraint is not recommended in production environments.
```

**Pull request merged**